src/Verify/IoHelpers.cs

```diff
@@ -201,8 +201,15 @@ public static string GetRelativePath(string directory, string file)
     private static VirtualizedRunHelper GetForAssembly(Assembly assembly) =>
         virtualizedRunHelpers.GetOrAdd(assembly, a => new(a));
 
-    public static string? GetDirectoryName(string? path) =>
-        Path.GetDirectoryName(GetMappedBuildPath(path));
+    public static string? GetDirectoryName(string? path)
+    {
+        var mappedPath = GetMappedBuildPath(path);
+        // Path remapping might not always be working (ex: you've built your project on Windows and you're running tests
+        // on the same assemblies but on another Linux machine). Always use / so that Path.GetDirectoryName does not
+        // return null on Unix systems.
+        mappedPath = mappedPath?.Replace("\\", "/");
```

Comment:
If we want to handle x-platfrom, then we should probably handle both directions (so e,g, as well a unix built binaries called on win).

so something like

```C#
static readonly char[] separators =
{
    '\\',
    '/'
};

// ...

if (!string.IsNullOrEmpty(mappedPath) && !mappedPath.Contains(Path.DirectorySeparatorChar))
{
  char contraSeparator = separators.First(c => !c.Equals(Path.DirectorySeparatorChar));
  mappedPath = mappedPath.Replace(contraSeparator, Path.DirectorySeparatorChar);
}
```